snake.asm

```diff
@@ -1,42 +1,42 @@
-push 0xB800
-pop ds
-mov cx, 0xFA0
-std
-start:
-	mov ax, 0x3
-	int 0x10
-	mov di, 0x7D0
-	mov si, sp
-.food:
-	imul bx, sp
-	and bx, cx
-	cmp [bx], ch
-	je .food
-	mov [bx], cl
-.input:
-	in al, 0x60
-	mov bx, 0x4
-	and al, 0x1E
-	jp $+0x4
-	mov bl, cl
-	and al, 0x14
-	jz $+0x4
-	neg bx
-	sub di, bx
-	cmp di, cx
-	ja start
-	sar bx, 0x1
-	lea ax, [di+bx+0x2]
-	div cl
-	and ah, ah
-	jz start
-	cmp [di], ch
-	je start
-	push di
-	cmp [di], cl
-	mov [di], ch
-	je .food
-	es lodsw
-	xchg ax, bx
-	mov [bx], ah
-	jmp SHORT .input
+push 0xB800                 ; push the offset to video memory for DMA
+pop ds                      ; pop that value into the data segment register so that every memory access will edit the video memory
+mov cx, 0xFA0               ; move the size of the screen in text mode 3 into cx, also the cp437 values of the food and snake body into ch, cl, and the width of the screen to cl
+std                         ; set direction flag to inverse the direction of lodsw for reading the stack as FIFO
+start:                      ; start new game setup
+	mov ax, 0x3         ; set ah=0 (change video mode) and al=3 (80x25 16 color text mode)
+	int 0x10            ; video BIOS interrupt
+	mov di, 0x7D0       ; move the middle of the screen (0xFA0/2) into di which is the initial position for the snake
+	mov si, sp          ; make si point to the stack by moving the value of the stack pointer into it so that every lods will access the stack
+.food:                      ; print new food
+	imul bx, sp         ; multiply bx by sp to pseudo randomize bx which points to the place we'll put the food
+	and bx, cx          ; and bx with cx to make it divisible by 4 (so it'll be eatable by the snake) and fit on the screen
+	cmp [bx], ch        ; check if bx points to a place on the screen with a snake body character 
+	je .food            ; if it is try to get another random bx
+	mov [bx], cl        ; else put a food character there
+.input:                     ; process keyboard input
+	in al, 0x60         ; scan port 60h (keyboard input)
+	mov bx, 0x4         ; set the default distance to move the snake to 4 (one step leftwards)

```

Comment:
Everything that happens below makes me think you might want another multiline comment above this section to explain how the snake moves in relation to the display and coordinates.